### ISSUE TITLE
Refered column creation more visible

### ISSUE THREAD

**Issue by CamilleLegeron** [gristlabs/grist-core#429]
Hello

By doing user tests, we realized that there is a not clear behavior. Users often don't understand / see that if a referenced data doesn't exist it must be explicitly created. 

![Peek 2023-02-14 16-22](https://user-images.githubusercontent.com/19594405/218782822-bf8509d9-0ce1-4fb3-af13-2d370334cb0a.gif)

Instand of use the `+` button they click on Enter and not understand why the field is red, it is not clear that the referenced data have not been created
![image](https://user-images.githubusercontent.com/19594405/218782116-b4241dc0-b444-4999-b8ee-88a8b07122da.png)

Suggestions : 
- We can bring the button to top of list (risk : the user could create a new referenced data before check if already exists)
- Remove element not corresponding to search entry (so the `+` button will remain the only one visible)
- Consider enter as a creation (maybe it's risky if there is a miss click, dirty data can be created)

I think I prefer the second suggestion. Want do you think ?

**Comment by vviers** [comment]
FWIW my vote goes to option number 2 though I suspect that the way reference fields search works is non-trivial 🤔

**Comment by dsagal** [comment]
I agree that option 2 sounds best of the ones listed (to show only the "+" item when no other choices match).

Another option (not mutually exclusive) is to pop up a dialog when a user hits Enter on a value that doesn't match anything. It could offer the choices of
- Create new record in **Table1** with `Field Name` of "CMS New", and add a reference to it.
- Store "CMS New" as text (not a valid reference).
- Cancel (revert change).

**Comment by CamilleLegeron** [comment]
@anaisconce It may require a little bit of UX/UI design, right ?

**Comment by anaisconce** [comment]
Option 2 is sensible! 

As for Dmitry's suggestion, what if instead of a pop-up, there's a one-click way to add the reference and fix the broken link. I am picturing something like this. 

![image](https://user-images.githubusercontent.com/69366135/221009747-a4e89a8d-4af1-4a72-ada4-06582422b8cf.png)
 Clicking the green `+` icon would be the same as double clicking into the cell to add it from the menu. On hover, the green icon in the red cell can darken to the darker green color, as it does when adding from the reference search list, in order to better communicate that this is a button and not just an icon.

![image](https://user-images.githubusercontent.com/69366135/221009548-079be028-1a8a-4573-82e8-ccf21c73f169.png)
 

A pop-up when hitting enter may be a bit jarring, it can feel like something is (disproportionately) wrong, or interrupt a workflow. The downside to using the small `+` button in the cell is that we don't seize the opportunity to teach a bit more about reference columns. But sometimes, for end users, that's the goal -- to make it easy and frictionless as if working in a custom app without having to parse too much information and make additional decisions.

Also I want it for myself because I often forget to click the `+` in the reference search list and then it's more clicks to fix that error. 😊

**Comment by mmarchand** [comment]
a user reported a UX issue to me about this,
when copy/pasting data in a table with references, it would be great to have a way to automatically add new entries in the reference table, 
there are probably different ways to handle this, could be a button to transform all red entries of the column into new entries in the column settings maybe or change the popup behavior here to make it default (maybe it needs a separate issue for this...)
